woocommerce/woocommerce-gateway-stripe
# Remove Stripe keys from the DB when uninstalling the plugin

Fixes #3384

## Changes proposed in this Pull Request:

This PR updates the `uninstall.php` to always remove the Stripe keys when uninstalling the plugin/

It also updates the list of other `options` to remove when the `WC_REMOVE_ALL_DATA` constant is true.

## Testing instructions

NOTE: You should test this on new local site (or a JN site), not your dev env; otherwise, the "Delete" option will remove your working copy.

1. Build a package from this branch, or use this zip: [woocommerce-gateway-stripe.zip](https://github.com/user-attachments/files/16969143/woocommerce-gateway-stripe.zip)
2. Go to the Plugin Settings page:
    `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`
3. Connect your Stripe account
4. Go to the Plugins page:
    `/wp-admin/plugins.php`
5. `Deactivate` and then `Delete` the `WooCommerce Stripe Gateway` plugin
6. Check that the plugin settings are still in the DB, for example using `wp-cli`:
    You can list all settings with:
    `wp option get woocommerce_stripe_settings`
7. Check that the Stripe API Keys were removed from the DB, for example using `wp-cli`
    You can list all settings with:
    `wp option get woocommerce_stripe_settings`
    Or individually check particular keys with:
    `wp option get woocommerce_stripe_settings`
    ![Screenshot 2024-09-02 at 4 50 20 PM](https://github.com/user-attachments/assets/0603a613-257b-4fba-bcf8-c5083e5f04af)
    For live mode, the keys are: `secret_key`, `publishable_key`, and `refresh_token` (only if using a Stripe App Connection), and for test mode: `test_secret_key`, `test_publishable_key`, and `test_refresh_token` (only if using a Stripe App Connection)
8. Install the Stripe Plugin again (you can use the same zip package)
9. Add the constant `WC_REMOVE_ALL_DATA` into your `wp-config.php` file:
    `define( 'WC_REMOVE_ALL_DATA', true );`
10. Connect your Stripe account
11. Go to the Plugins page:
    `/wp-admin/plugins.php`
12. `Deactivate` and then `Delete` the `WooCommerce Stripe Gateway` plugin.
13. Check that all the plugin settings were removed from the DB, for example using `wp-cli`:
    `wp option get woocommerce_stripe_settings`
    ![Screenshot 2024-09-11 at 4 36 37 PM](https://github.com/user-attachments/assets/a1745572-a074-4885-964c-8a5c132ce7a2)

---


**Post merge**
